shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ShardingSphereResultSet.java

```diff
@@ -367,8 +369,47 @@ public Object getObject(final String columnLabel) throws SQLException {
     @SuppressWarnings("unchecked")
     @Override
     public <T> T getObject(final int columnIndex, final Class<T> type) throws SQLException {
-        if (LocalDateTime.class.equals(type) || LocalDate.class.equals(type) || LocalTime.class.equals(type)) {
+        if (null == type) {
+            throw new SQLException("Type cannot be null");
+        }
+        if (String.class.equals(type)) {
```

Comment:
@RaigorJiang Can we move this logic to ResultSetUtil#convertValue, it seems to be the utility class responsible for type conversion.